app/models/validations/shared_validations.rb

```diff
@@ -34,6 +34,35 @@ def validate_numeric_min_max(record)
     end
   end
 
+  def validate_numeric_step(record)
+    record.form.numeric_questions.each do |question|
+      next unless question.step
+      next unless record[question.id] && question.page.routed_to?(record, nil)
```

Comment:
hmm. I might try and have a little experiment with this. What you say makes sense, I wonder if there's a chance because we're in a before-save lifecycle call back if there might be funny edge cases.
Really here, I copied the format of the above method for validate_numeric_min_max so I could get started and didn't come back round to question this.
Now I look in git blame it looks like you last touched the corresponding line on l21 above, maybe you can enlighten me as to why I need to do this! I will see what happens if I get rid of it

Reply:
hmmm. I am happy to remove this if you think it's fine, I wonder if there's an edge case where it makes sense though.
A page can be not routed to if the subsection it is in is not enabled, and perhaps that answer could be derived? I should hope that we don't derive invalid values, but maybe it's not crazy to have an extra safeguard that a validation error gets added to a question that the user can't even get to? although in that case we will be calling a value from the db that has already passed this validation, so I really can't imagine a case where this is necessary.

OK I see that you did add it in the other method and discussed it with sam in review
https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/1249/files#diff-86ebd101437a30c8df76e2e300c9c61a9cdecefd09e7594fe02cad46e7234aa2
maybe when you're free you can call/message if you remember that thing and whether it applies here (I don't tihnk I have the context for that conversation)